beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java

```diff
@@ -122,6 +122,9 @@ public class ValidatorApiHandler implements ValidatorApiChannel {
    */
   private static final int DUTY_EPOCH_TOLERANCE = 1;
 
+  private final Map<UInt64, SafeFuture<Optional<BlockContainerAndMetaData>>>
+      blockProductionFuturesBySlotCache = LimitedMap.createSynchronizedLRU(2);
+
   private final Map<UInt64, Bytes32> createdBlockRootsBySlotCache =
       LimitedMap.createSynchronizedLRU(2);
 
```

Comment:
I think we can get the rid of `createdBlockRootsBySlotCache` and use only the new `blockProductionFuturesBySlotCache`.